### PR TITLE
feat(port,content): Winged zombies

### PR DESCRIPTION
**Pull request by Zlorthishen**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change
ports 
 - https://github.com/CleverRaven/Cataclysm-DDA/pull/43188

### CODE REVIEW

File: data/json/monster_special_attacks/spells.json [hunk 1]
```diff
@@ -215,5 +215,69 @@
     "max_aoe": 3,
     "effect": "summon",
     "effect_str": "mon_centipede_small"
+  },
+  {
+    "type": "SPELL",
+    "id": "small_raptor_spawn",
+    "name": { "str": "Spawn Raptors" },
+    "description": "Summons several zombiespawn raptors.",
+    "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT" ],
```
Comment (chaosvolt) [inline data/json/monster_special_attacks/spells.json:224]:
```suggestion
    "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT", "NO_EXPLOSION_VFX", "SILENT" ],
```

File: data/json/monster_special_attacks/spells.json [hunk 2]
```diff
@@ -215,5 +215,69 @@
...
+    "extra_effects": [ { "id": "spawning_zombie_die", "hit_self": true } ],
+    "effect": "summon",
+    "effect_str": "mon_spawn_raptor"
+  },
+  {
+    "type": "SPELL",
+    "id": "big_raptor_spawn",
+    "name": { "str": "Spawn Raptors" },
+    "description": "Summons many zombiespawn raptors.",
+    "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT" ],
```
Comment (chaosvolt) [inline data/json/monster_special_attacks/spells.json:242]:
```suggestion
    "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT", "NO_EXPLOSION_VFX", "SILENT" ],
```

File: data/json/monster_special_attacks/spells.json [hunk 3]
```diff
@@ -215,5 +215,69 @@
...
+    "extra_effects": [ { "id": "spawning_zombie_die", "hit_self": true } ],
+    "effect": "summon",
+    "effect_str": "mon_spawn_raptor"
+  },
+  {
+    "type": "SPELL",
+    "id": "shady_raptor_spawn",
+    "name": { "str": "Spawn Shady Raptors" },
+    "description": "Summons a few shady zombiespawn raptors.",
+    "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT" ],
```
Comment (chaosvolt) [inline data/json/monster_special_attacks/spells.json:260]:
```suggestion
    "flags": [ "HOSTILE_SUMMON", "RANDOM_DAMAGE", "PERMANENT", "NO_EXPLOSION_VFX", "SILENT" ],
```

File: data/json/monster_special_attacks/spells.json [hunk 4]
```diff
@@ -215,5 +215,69 @@
...
+    "extra_effects": [ { "id": "spawning_zombie_die", "hit_self": true } ],
+    "effect": "summon",
+    "effect_str": "mon_spawn_raptor_shady"
+  },
+  {
+    "type": "SPELL",
+    "id": "spawning_zombie_die",
+    "name": { "str": "Spawning Zombie Die" },
+    "description": "For zombies that are spawning something and die in the process.",
+    "flags": [ "LOUD", "RANDOM_DAMAGE" ],
```
Comment (chaosvolt) [inline data/json/monster_special_attacks/spells.json:276]: Random damage isn't needed when it's 250 for min and max.
```suggestion
    "flags": [ "NO_EXPLOSION_VFX", "SILENT" ],
```

**Review by chaosvolt — COMMENTED** [review]
Looks aight, they no longer spawn next to the player which was if I recall what killed the port the first time.

These flags would be good to have to make it not make explosion noises and show a visual explosion effect though.

**Review by chaosvolt — APPROVED** [review]

**Comment by chaosvolt** [comment]
Wait, is it intentional that they can't actually fly? o.O

**Review by chaosvolt — COMMENTED** [review]
So, some issues with this I spotted a bit belatedly:
1. Winged zombies and garghouls can't fly, only flesh raptors can. I guess that makes sense if you're being realismic, but a side effect of that is that winged zombies and garghouls don't really have any reason to exist. But funny thing about that...
2. That's the neat part, they don't! Crawlers were given pupas as evolutions so those will spawn, but batwing zombies and garghouls don't have any spawns or evolution connections.

I'd say give winged zombies monstergroup inclusions with a `starts` delay of about 500, since they're potentially bigger and beefier than flesh raptors that have a half-life of 14 (which should work out to 336 hours), but in exchange add `FLIES` to them and garghouls, and adjust descriptions where needed.

**Review by chaosvolt — APPROVED** [review]

**Comment by NappingOcean** [comment]
I think you forgot the monsters that evolve into `mon_zombie_pupa`. Maybe `mon_zombie_fat`?

**Comment by chaosvolt** [comment]
Oh yeah, didn't notice that earlier, hecc.

**Review by chaosvolt — CHANGES_REQUESTED** [review]
Things I once again didn't spot last time:
1. `GROUP_ZOMBIE_RAPTOR_UPGRADES` is unused. I assume it's meant to be an evolution for any flesh raptors that spawn? If so you'd want to change the default, I doubt it will like a recursive evolution. Alternatively you could give it as an evolution to pupas to allow them to burst off-map over time, and this would make the existing default monster make more sense (as then it'd mean pupas that spawn later on get replaced with a wide variety of raptors).
2. `GROUP_ZOMBIE_SHADY_UPGRADES` is unimplemented. I assume it's meant to be used as an evolution for shady zombies? If so again, default is already `mon_zombie_shady` so you'd likely want to tweak that.
3. As noted, `mon_zombie_pupa_decoy` and `mon_zombie_pupa` don't spawn. Placing them in `GROUP_ZOMBIE_FAT` may be reasonable.

**Review by chaosvolt — APPROVED** [review]
image

**Pull request merged**